sdk/tests/conformance2/rendering/framebuffer-render-to-layer.html

```diff
@@ -0,0 +1,461 @@
+      v[gl.RGB_INTEGER]  = [1, 1, 1, 0];
+      v[gl.RGBA_INTEGER] = [1, 1, 1, 1];
+  }
+
+  const depthTextureFormats = [
+      gl.DEPTH_COMPONENT16,
+      gl.DEPTH_COMPONENT24,
+      gl.DEPTH_COMPONENT32F,
+      gl.DEPTH24_STENCIL8,
+      gl.DEPTH32F_STENCIL8,
+  ];
+
+  const intTextureFormats = [
+      gl.R8I,
+      gl.R16I,
+      gl.R32I,
+      gl.RG8I,
+      gl.RG16I,
+      gl.RG32I,
+      gl.RGB8I,
+      gl.RGB16I,
+      gl.RGB32I,
+      gl.RGBA8I,
+      gl.RGBA16I,
+      gl.RGBA32I,
+  ];
+
+  const unsignedIntTextureFormats = [
+      gl.R8UI,
+      gl.R16UI,
+      gl.R32UI,
+      gl.RG8UI,
+      gl.RG16UI,
+      gl.RG32UI,
+      gl.RGB8UI,
+      gl.RGB16UI,
+      gl.RGB32UI,
+      gl.RGBA8UI,
+      gl.RGB10_A2UI,
+      gl.RGBA16UI,
+      gl.RGBA32UI,
+  ];
+
+  const floatTextureFormats = Object.keys(textureInternalFormatInfo).map(function(v) {
+      return parseInt(v);
+  }).filter(function(format) {
+      return intTextureFormats.indexOf(format) < 0 &&
+             unsignedIntTextureFormats.indexOf(format) < 0 &&
+             depthTextureFormats.indexOf(format);
+  });
+
+  const expectedColorByInternalFormat = {};
+  expectedColorByInternalFormat[gl.SRGB8_ALPHA8] = [225, 188, 137, 255];
+
+  function clearFloat(gl) {
+      gl.clearBufferfv(gl.COLOR, 0, [0, 0, 0, 0]);
+  }
+
+  function clearInt(gl) {
+      gl.clearBufferiv(gl.COLOR, 0, [0, 0, 0, 0]);
+  }
+
+  function clearUint(gl) {
+      gl.clearBufferuiv(gl.COLOR, 0, [0, 0, 0, 0]);
+  }
+
+
+  function checkData(data, expected, internalFormat, tolerance) {
+      const internalFormatInfo = textureInternalFormatInfo[internalFormat];
+      const validChannels = validChannelsByTextureFormat[internalFormatInfo.textureFormat];
+      if (!validChannels) {
+          testFailed('oops');
+          return;
+      }
+      for (let y = 0; y < texHeight; ++y) {
+          for (let x = 0; x < texWidth; ++x) {
+              for (let c = 0; c < validChannels.length; ++c) {
+                  if (validChannels[c]) {
+                      const offset = (y * texWidth + x) * 4 + c;
+                      const pixel = data[offset];
+                      const diff = Math.abs(pixel - expected[c]);
+                      if (diff > tolerance) {
+                          testFailed(`pixel ${x},${y} channel ${c} was ${pixel} expected ${expected[c]}`);
+                          return;
+                      }
+                  }
+              }
+          }
+      }
+      testPassed(`data was ${expected.join(',')}`);
+  }
+
+  function checkFloat(gl, textureInfo, expected) {
+      const data = new Uint8Array(texWidth * texHeight * 4);
+      gl.readPixels(0, 0, texWidth, texHeight, gl.RGBA, gl.UNSIGNED_BYTE, data);
+      const internalFormat = textureInfo.internalFormat;
+      wtu.glErrorShouldBe(gl, gl.NO_ERROR, `No errors from readPixels with ${wtu.glEnumToString(gl, internalFormat)}`);
+      checkData(data, expected, internalFormat, 5);
```

Comment:
5 isn't enough for RGBA4, and it fails for me. This worked: (136 is what I get for RGBA4)
```
const tolerance = 136-127;
checkData(data, expected, internalFormat, tolerance);
```

Alternatively, consider having a per-format tolerance.